src/server/pfs/server/obj_block_api_server.go

```diff
@@ -155,29 +187,20 @@ func (s *objBlockAPIServer) PutBlock(putBlockServer pfsclient.BlockAPI_PutBlockS
 
 func (s *objBlockAPIServer) GetBlock(request *pfsclient.GetBlockRequest, getBlockServer pfsclient.BlockAPI_GetBlockServer) (retErr error) {
 	defer func(start time.Time) { s.Log(request, nil, retErr, time.Since(start)) }(time.Now())
-	var reader io.ReadCloser
-	var err error
-	backoff.RetryNotify(func() error {
-		reader, err = s.objClient.Reader(s.localServer.blockPath(request.Block), request.OffsetBytes, request.SizeBytes)
-		if err != nil && s.objClient.IsRetryable(err) {
-			return err
-		}
-		return nil
-	}, obj.NewExponentialBackOffConfig(), func(err error, d time.Duration) {
-		protolion.Infof("Error creating reader; retrying in %s: %#v", d, obj.RetryError{
-			Err:               err.Error(),
-			TimeTillNextRetry: d.String(),
-		})
-	})
-	if err != nil {
+	var data []byte
+	sink := groupcache.AllocatingByteSliceSink(&data)
+	if err := s.cache.Get(getBlockServer.Context(), request.Block.Hash, sink); err != nil {
 		return err
 	}
-	defer func() {
-		if err := reader.Close(); err != nil && retErr == nil {
-			retErr = err
-		}
-	}()
-	return protostream.WriteToStreamingBytesServer(reader, getBlockServer)
+	if request.SizeBytes != 0 && request.SizeBytes+request.OffsetBytes < uint64(len(data)) {
+		data = data[request.OffsetBytes : request.OffsetBytes+request.SizeBytes]
+	} else if request.OffsetBytes < uint64(len(data)) {
+		data = data[request.OffsetBytes:]
+	} else {
+		data = nil
+	}
+	getBlockServer.Send(&google_protobuf.BytesValue{Value: data})
```

Comment:
I think you just want to return this value. Otherwise if it errs, we don't report it